### PR TITLE
Simplified code for thread synchronization

### PR DESCRIPTION
**Pull request by emilwestergren**
@marcusnaslund what do you think of this?

### CODE REVIEW

**Comment by marcusnaslund** [comment]
Much better. Except the rock segfaulting.

**Comment by emilwestergren** [comment]
Just add it to my PR debt

**Comment by emilwestergren** [comment]
Let's begin with this.

File: source/concurrent/ThreadPool.ooc [hunk 1]
```diff
@@ -37,6 +37,18 @@ _Task: abstract class {
...
 		this _state == _PromiseState Finished
 	}
+	wait: func ~timeout (time: TimeSpan) -> Bool {
+		timer := CpuTimer new() . start()
+		seconds := time toSeconds()
+		status := false
+		while (timer stop() / 1000.0 < seconds && !status) {
+			status = (this _state != _PromiseState Unfinished)
+			if (!status)
+				Time sleepMilli(seconds / 10 as Int)
```
Comment (marcusnaslund) [inline source/concurrent/ThreadPool.ooc]: `sleepMilli` with seconds argument?
Oh, it was like that before, but, why like this?

Comment (emilwestergren) [inline source/concurrent/ThreadPool.ooc]: No idea, I'm changing this in a later commit.

**Comment by marcusnaslund** [comment]
Looks good.

**Pull request merged**